core/src/test/java/org/infinispan/partitionhandling/BaseTxPartitionAndMergeTest.java

```diff
@@ -289,16 +291,18 @@ public void unblock() {
 
       private final Class<? extends CacheRpcCommand> aClass;
       private final ReclosableLatch notifier;
+      private volatile boolean discard;
 
       private DiscardFilter(Class<? extends CacheRpcCommand> aClass) {
          this.aClass = aClass;
          notifier = new ReclosableLatch(false);
+         discard = true; //discard everything by default. change if needed in the future.
```

Comment:
@rvansa I guess the key is that now everything is discarded by default, before a command would be discarded only when `!notifier.isOpened()`

Reply:
that was the problem. only the first command is was discarded. The test expects the command only to be executed during the partition merging. So, I discard everything (only a single operation is executed in the tests) and I stop discarding when I merge